framework/src/main/java/com/github/dbmdz/flusswerk/framework/rabbitmq/RabbitClient.java

```diff
@@ -52,82 +91,114 @@ void send(String exchange, String routingKey, Message message) throws IOExceptio
     sendRaw(exchange, routingKey, data);
   }
 
-  void sendRaw(String exchange, String routingKey, byte[] data) throws IOException {
+  void sendRaw(String exchange, String routingKey, byte[] data) {
     AMQP.BasicProperties properties =
         new AMQP.BasicProperties.Builder()
             .contentType("application/json")
             .deliveryMode(PERSISTENT)
             .build();
 
-    try {
-      channel.basicPublish(exchange, routingKey, properties, data);
-    } catch (Exception e) {
-      tryToReconnect("Could not publish message to " + routingKey);
-      channel.basicPublish(exchange, routingKey, properties, data);
+    // The channel might not be available or become unavailable due to a connection error. In this
+    // case, we wait until the connection becomes available again.
+    while (true) {
+      if (channelAvailable) {
+        try {
+          channel.basicPublish(exchange, routingKey, properties, data);
+          break;
+        } catch (IOException | AlreadyClosedException e) {
+          log.warn("Failed to publish message to RabbitMQ: {}", e.getMessage(), e);
+          channelAvailable = false;
+        }
+      }
+      // We loop here because the signal might be triggered due to what the JVM documentation calls
+      // a 'spurious wakeup', i.e. the signal is triggered even though no connection recovery has
+      // yet happened.
+      while (!channelAvailable) {
+        channelLock.lock();
+        channelAvailableAgain.awaitUninterruptibly();
+        channelLock.unlock();
+      }
```

Comment:
Could we wait for the condition `channelAvailableAgain` in the catch-block instead of adding another loop?

Reply:
No, because what happens when the channel is not available to begin with, i.e. the user calls `sendRaw` when the channel is already closed?
In this case the exception handler is never run, and without this additional loop where we wait on the signal, we would enter a busy loop until the channel becomes available again.